game/source_engine_nav.ksy

```diff
@@ -0,0 +1,467 @@
+  - id: ladder_count
+    type: u4
+    if: '_root.version >= 6'
+    doc: Amount of ladders stored in the navigation mesh. Added in NAV version 6.
+  - id: ladder_array
+    type: nav_ladder
+    repeat: expr
+    repeat-expr: ladder_count
+    if: '_root.version >= 6'
+  - id: custom_data
+    size-eos: true
+    if: '_root.version >= 10 and _root.subversion != 0'
+    doc: Custom data appended by a specialized implementation of the NAV format. Should only be parsed if NAV file has a subversion.
+
+types:
+  place_name:
+    doc: |
+      A label of a location in a map. They are defined by the map (BSP file). Added in NAV Version 5.
+
+      Place names are stored as Length-String value pairs.
+    seq:
+      - id: length
+        type: u2
+        doc: Length of the place name string.
+      - id: name
+        type: str
+        size: length
+        encoding: ASCII
+  # Connection Pair
+  area_id_array:
+    seq:
+      - id: count
+        type: u4
+      - id: area_ids
+        type: u4
+        repeat: expr
+        repeat-expr: count
+  # ApproachInfo
+  approach_info:
+    doc: |
+      Datum point that determines the type of movement between areas. Removed in NAV version 15.
+
+      It is only used by the Counter-Strike video game series.
+
+    # The elements
+    seq:
+      - id: target_area_id
+        type: u4
+        doc: Area ID of area to approach to.
+      - id: approach_prev
+        type: u4
+        doc: Area ID of the area before the current approach area.
+      - id: type
+        type: u1
+        doc: Type of approach method.
+      - id: next
+        type: u4
+        doc: Area ID of area to approach to next.
+      - id: method
+        type: u1
+  # NavHidingSpot
+  hiding_spot:
+    seq:
+      - id: area_id
+        type: u4
+        if: '_root.version >= 2'
+        doc: Area ID of the hiding spot's parent area. Added in NAV version 2.
+      - id: position
+        type: f4
+        repeat: expr
+        repeat-expr: 3
+        if: '_root.version >= 1'
+        doc: The position of the Hiding spot (as a vector). Added in NAV version 1.
+      - id: attribute_flag
+        type: b1
+        repeat: expr
+        repeat-expr: 8
+        if: '_root.version >= 2'
+        doc: The hiding spot attribute flag. Added in NAV version 2.
+    enums:
+      # Hiding Spot Attributes.
+      hide_spot_attribute:
+        1: in_cover
+        2: good_sniper_spot
+        4: ideal_sniper_spot
+        8: exposed
+  # Class: AreaBindInfo
+  area_bind:
+    doc: |
+      Stores the type of visibility to an area. Added in NAV version 16.
+
+      The structure stores an area ID as an unsigned integer (32-bits) and a byte flag that determines if visibility checks should be done.
+    seq:
+      - id: bound_area_id
+        type: u4
+        doc: Identifier of the targeted area.
+      - id: attribute_byte
+        type: u1 # Value is used as a number, NOT as a flag.
+        enum: visibility_type
+        doc: |
```

Comment:
Move these docs into `visibility_type` items (
```yaml
visibility_type:
  0:
    id: not_visible
    doc: '\No visibility checks; area is always invisible.
```
) and set `enum: visibility_type` here.

Reply:
Done.